carma_novatel_driver_wrapper/src/carma_novatel_driver_wrapper.cpp

```diff
@@ -0,0 +1,207 @@
+        imu_sub_ = create_subscription<sensor_msgs::msg::Imu>("imu_raw", 5,
+            std::bind(&CarmaNovatelDriverWrapper::imu_callback, this, std::placeholders::_1));
+
+        fix_fused_pub_ = create_publisher<gps_msgs::msg::GPSFix>("gnss_fix_fused", 10.0);
+        status_pub_= create_publisher<carma_driver_msgs::msg::DriverStatus>("driver_discovery",10);
+        alert_pub_ = create_publisher<carma_msgs::msg::SystemAlert>("system_alert",10);
+
+    }
+
+    void CarmaNovatelDriverWrapper::inspvax_callback(const novatel_oem7_msgs::msg::INSPVAX::UniquePtr msg)
+    {
+        last_imu_msg_ = this->now();
+
+        gps_msgs::msg::GPSFix fix_msg;
+        // NOTE: At the moment CARMA's hardware interfaces do not use GPS status information
+        // Therefore that information is not currently populated in this message
+        // Convert position
+        fix_msg.header = msg->header;
+        fix_msg.latitude = msg->latitude;
+        fix_msg.longitude = msg->longitude;
+        fix_msg.altitude = msg->height;
+        //Set position covariance
+        fix_msg.position_covariance_type = gps_msgs::msg::GPSFix::COVARIANCE_TYPE_DIAGONAL_KNOWN;
+        fix_msg.position_covariance[0] = msg->longitude_stdev * msg->longitude_stdev;
+        fix_msg.position_covariance[4] = msg->latitude_stdev * msg->latitude_stdev;
+        fix_msg.position_covariance[8] = msg->height_stdev * msg->height_stdev;
+
+        //Convert orientation
+        // NOTE: It is unclear in the message spec what dip represents so it will be ignored
+        fix_msg.track = msg->azimuth;
+        fix_msg.pitch = msg->pitch;
+        fix_msg.roll = msg->roll;
+
+        // GPSFix messages request uncertainty reported with 95% confidence interval. That is 2 times the standard deviation
+        fix_msg.err_track = 2.0 * msg->azimuth_stdev;
+        fix_msg.err_pitch = 2.0 * msg->pitch_stdev;
+        fix_msg.err_roll = 2.0 * msg->roll_stdev;
+
+        // Convert Velocity
+        std::vector<double> components, variances;
+        components.push_back(msg->north_velocity);
+        variances.push_back(2.0 * msg->north_velocity_stdev);
+
+        components.push_back(msg->east_velocity);
+        variances.push_back(2.0 * msg->east_velocity_stdev);
+
+        // std::tuple<double, double> speed_tuple = uncertainty_tools::computeVectorMagnitudeAndUncertainty(components, variances);
+        //fix_msg.speed = std::get<0>(speed_tuple);
+        fix_msg.climb = msg->up_velocity;
+
+        fix_msg.err_climb = 2.0 * msg->up_velocity_stdev;
+        //fix_msg.err_speed = std::get<1>(speed_tuple);
+
+        fix_fused_pub_->publish(fix_msg);
+
+    }
+
+    void CarmaNovatelDriverWrapper::imu_callback(const sensor_msgs::msg::Imu::UniquePtr msg)
+    {
+        last_gnss_msg_ = this->now();
+    }
+
+    void CarmaNovatelDriverWrapper::status_timer_callback(const rclcpp::TimerCallbackType& event)
```

Comment:
Driver discovery behavior is not needed (thus why it was not included in the relevant interfaces)